python/ctranslate2/converters/transformers.py

```diff
@@ -627,6 +627,98 @@ def set_decoder(self, spec, module, rotary_dim, num_heads):
             self.set_linear(layer_spec.ffn.linear_1, layer.mlp.fc_out)
 
 
+@register_loader("CodeGenConfig")
+class CodeGenLoader(ModelLoader):
+    @property
+    def architecture_name(self):
+        return "CodeGenForCausalLM"
+
+    def get_model_spec(self, model):
+        spec = transformer_spec.TransformerDecoderModelSpec.from_config(
+            model.config.n_layer,
+            model.config.n_head,
+            pre_norm=True,
+            activation=_SUPPORTED_ACTIVATIONS[model.config.activation_function],
+            rotary_dim=model.config.rotary_dim,
+            rotary_interleave=False,
+            parallel_residual=True,
+            shared_layer_norm=True,
+        )
+
+        mp_num = 4
+        if (
+            "Salesforce/codegen2-1B" in model.name_or_path
+            or "Salesforce/codegen2-3_7B" in model.name_or_path
+        ):
+            # consider mp_num=8 for smaller codegen2 series.
+            mp_num = 8
```

Comment:
Is there a way to avoid checking the model name here? Can this value be set based on the model dimension for example (`n_embd` in the configuration)?

Reply:
Codegen2-1B: 2048 => mp_num=8 (via trust_remote_code=True)
Codegen2-3_7B: n_embd: 4096,=> mp_num=8 (via trust_remote_code=True)
Codegen2-7B: n_embd: 4096 => mp_num=4
Codegen2-16B: n_embd: 6144=> mp_num=4

Even worse:
- Salesforce/codegen-6B-mono: n_embd: 6144=> mp_num=4, which also uses `CodeGenForCausalLM` and the same config.

Only difference would be to consider a argument called "head_dim" in 
 "Salesforce/codegen2-1B""Salesforce/codegen2-3_7B", which is not present in the other.
 I am not sure, if thats a smart choice either, since 'head_dim' is just legacy, redundant information (head_dim==n_emb/n_head), and is not related to mp_num.
 
There is one big benefit, when somebody forks the existing code, this would rename the model_name.
Ill change it and probe if "head_dim" exists and equals to 128 or 256.

Reply:
Ok, thanks for the info. It seems better to check the head dim for now.